photon-client/src/integTest/scala/com/linkedin/photon/ml/cli/game/training/DriverTest.scala

```diff
@@ -447,6 +449,43 @@ class DriverTest extends SparkTestUtils with GameTestUtils with TestTemplateWith
     }
   }
 
+  @DataProvider(name = "badWeightsInputs")
+  def badWeightsInputs(): Array[Array[String]] = {
+
+    val inputDir = getClass.getClassLoader.getResource("DriverIntegTest/input/bad-weights/").getPath
+    new File(inputDir)
+      .listFiles
+      .filter(_.getName.endsWith(".avro"))
+      .map(f => Array(inputDir + "feature-lists", f.getAbsolutePath))
+  }
+
+  /**
+   * Make sure we filter out negative or zero sample weights, and throw an exception if there are no samples left after
+   * filtering.
+   */
+  @Test(dataProvider = "badWeightsInputs") //, expectedExceptions = Array(classOf[IllegalArgumentException]))
+  def testBadSampleWeights(featuresDir: String, inputFile: String): Unit = sparkTest("testBadSampleWeights") {
+
+    val args =
+      Map("task-type" -> TaskType.LINEAR_REGRESSION.toString,
+        "feature-name-and-term-set-path" -> featuresDir,
+        "feature-shard-id-to-feature-section-keys-map" -> "shard1:features",
+        "train-input-dirs" -> inputFile,
+        "fixed-effect-data-configurations" -> "global:shard1,2",
+        "fixed-effect-optimization-configurations" -> "global:100,1e-11,0.3,1,LBFGS,l2",
+        "updating-sequence" -> "global",
+        "validate-input-dirs" -> "",
+        "output-dir" -> s"$getTmpDir/output",
+        "num-iterations" -> "5",
+        "check-data" -> "true")
```

Comment:
You can probably replace most of these entries by reusing `fixedEffectToyRunArgs` or `fixedEffectSeriousRunArgs`